sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -752,7 +752,7 @@ nonReserved
     | UNBOUNDED | WHEN
     | DATABASE | SELECT | FROM | WHERE | HAVING | TO | TABLE | WITH | NOT
     | DIRECTORY
-    | BOTH | LEADING | TRAILING
+    | TRIM | BOTH | LEADING | TRAILING
```

Comment:
@wangyum . 
I agree your intention because this location might be better. However, we consider the ordering seriously during `backporting` .
`master` branch had better resemble `branch-2.4`. `branch-2.4` branch had better reusable `branch-2.3`. Please follow the nearest branch. In this case, `branch-2.4`. Please put the change to the closest line compared with `branch-2.4`.

Reply:
OK.  I see. Thank you @dongjoon-hyun